addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2562,7 +2583,11 @@ const RangeUserValueWidget = UnitUserValueWidget.extend({
         const inputValue = possibleValues.length > 1 ? possibleValues.indexOf(value) : this._value;
         this.input.value = inputValue;
         if (this.displayValue) {
-            this.outputEl.value = inputValue;
+            let inputValueAsNumber = Number(inputValue);
+            if (this.exponential) {
+                inputValueAsNumber = Math.exp(inputValueAsNumber);
+            }
+            this.outputEl.value = this.fixedDigits ? inputValueAsNumber.toFixed(this.fixedDigits) : inputValueAsNumber.toString();
```

Comment:
All this code must be in a function because it is duplicated in `_onInputInput`. This reduces the number of lines and is easier to maintain.